lib/matplotlib/widgets.py

```diff
@@ -1487,7 +1488,10 @@ def _press(self, event):
         self.rect.set_visible(self.visible)
         if self.span_stays:
             self.stay_rect.set_visible(False)
-
+            # really force a draw so that the stay rect is not in
+            # the blit background
+            if self.useblit:
+                self.canvas.draw()
```

Comment:
If the stay rectangle is present then the background that is grabbed will include it, by forcing the draw here it is removed from the background used for blitting.